### PR TITLE
Option  --install-prefix

### PR DESCRIPTION
**Pull request by MTCam**
This PR adds the capability to install _mirgecom_ (and its development dependency packages) to a user-specified directory.  It modifies the existing option `--prefix` = `--conda-prefix` and adds a new `--install-prefix` option.

### CODE REVIEW

File: fetch-mirgecom.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
```
Comment (matthiasdiener) [inline fetch-mirgecom.sh:2]:
```suggestion
set -o nounset -o errexit
```

File: fetch-mirgecom.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+mcbranch=$1
+mcprefix=$2
```
Comment (matthiasdiener) [inline fetch-mirgecom.sh:4]: Should add some usage help here

File: fetch-mirgecom.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# default it to branch=master, install=PWD
+if [ -z "$mcbranch" ]; then mcbranch="master"; fi
+if [ -z "$mcprefix" ]; then mcprefix=`pwd`; fi
```
Comment (matthiasdiener) [inline fetch-mirgecom.sh:7]:
```suggestion
[[ -z "$mcbranch" ]] && mcbranch="master"
[[ -z "$mcprefix" ]] && mcprefix=$(pwd)
```

File: fetch-mirgecom.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+# create or populate mirgecom from repo
+mcsrc=$mcprefix/mirgecom
+if [ -f $mcsrc/setup.py ]
```
Comment (matthiasdiener) [inline fetch-mirgecom.sh:10]:
```suggestion
if [[ -f "$mcsrc/setup.py" ]]
```

File: fetch-mirgecom.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+then   # mirgecom src already populated, checkout the right branch, pull it
+    cd ${mcsrc}&&git checkout ${mcbranch}&&git pull
```
Comment (matthiasdiener) [inline fetch-mirgecom.sh:12]:
```suggestion
    cd ${mcsrc} && git checkout ${mcbranch} && git pull
```

File: install-pip-dependencies.sh [hunk 6]
```diff
@@ -2,22 +2,32 @@
 
 set -o nounset -o errexit
```
Comment (matthiasdiener) [inline install-pip-dependencies.sh:4]: same as above, needs usage help

File: install-pip-dependencies.sh [hunk 7]
```diff
@@ -2,22 +2,32 @@
...
 
-
+requirements_file=$1
+install_location=$2
+
+if [ -z "$install_location" ]
```
Comment (matthiasdiener) [inline install-pip-dependencies.sh:8]:
```suggestion
if [[ -z "$install_location" ]]
```

File: install-pip-dependencies.sh [hunk 8]
```diff
@@ -2,22 +2,32 @@
...
+then
+    install_location=`pwd`
```
Comment (matthiasdiener) [inline install-pip-dependencies.sh:10]:
```suggestion
    install_location=$(pwd)
```

File: install-src-package.sh [hunk 9]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
```
Comment (matthiasdiener) [inline install-src-package.sh:2]:
```suggestion
#!/bin/bash

set -o nounset -o errexit
```

File: install-src-package.sh [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+packagepath=$1
+if [ -z "$packagepath" ]
```
Comment (matthiasdiener) [inline install-src-package.sh:4]:
```suggestion
if [[ -z "$packagepath" ]]
```

File: install-src-package.sh [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+then
+    printf "Error: No package path given.\n"
```
Comment (matthiasdiener) [inline install-src-package.sh:6]:
```suggestion
    echo "Error: No package path given."
```

File: install.sh [hunk 12]
```diff
@@ -4,22 +4,25 @@ set -o errexit
...
 {
-  echo "Usage: $0 [--prefix=DIR] [--branch=NAME] [--modules] [--help]"
-  echo "  --prefix=DIR      Install conda in non-default prefix."
-  echo "  --modules         Create modules.zip and add to Python path."
-  echo "  --branch=NAME     Install specific branch of mirgecom (default=master)."
-  echo "  --help            Print this help text."
+  echo "Usage: $0 [--install-prefix=DIR] [--conda-prefix=DIR] [--branch=NAME]" 
+  echo "                   [--modules] [--help]"
+  echo "  --install-prefix=DIR  Install mirgecom in [DIR], (default=PWD)."
+  echo "  --conda-prefix=DIR    Install conda in [DIR], (default=~/Miniconda3)"
```
Comment (matthiasdiener) [inline install.sh:16]:
```suggestion
  echo "  --conda-prefix=DIR    Install conda in [DIR], (default=~/miniforge3)"
```

File: install.sh [hunk 13]
```diff
@@ -4,22 +4,25 @@ set -o errexit
...
+  echo "  --modules             Create modules.zip and add to Python path."
+  echo "  --branch=NAME         Install specific branch of mirgecom (default=master)."
+  echo "  --help                Print this help text."
 }
 
 # Default conda location
 conda_prefix=$HOME/miniforge3
-mirgecom_branch="master"
+mcbranch="master"
+mcprefix=`pwd`
```
Comment (matthiasdiener) [inline install.sh:25]:
```suggestion
mcprefix=$(pwd)
```

File: install.sh [hunk 14]
```diff
@@ -70,17 +78,29 @@ conda create --name dgfem --yes
 
 source $MY_CONDA_DIR/bin/activate dgfem
 
+if [ ! -d $mcprefix ]
+then
+    mkdir -p $mcprefix
+fi
```
Comment (matthiasdiener) [inline install.sh:84]:
```suggestion
mkdir -p $mcprefix
```

**Comment by MTCam** [comment]
Shoulda batched those.  I see that's an option

**Comment by MTCam** [comment]
@matthiasdiener Do you have *any idea* how a '#' is getting into dependencies?  This wasn't happening before.

**Comment by matthiasdiener** [comment]
> @matthiasdiener Do you have _any idea_ how a '#' is getting into dependencies? This wasn't happening before.

Yes, it's a fallout from https://github.com/illinois-ceesd/mirgecom/pull/51 (and caused the current discussion in slack).

File: install-pip-dependencies.sh (hunk 7 above)
Comment (MTCam) [reply install-pip-dependencies.sh:17]: Out of curiosity, what do those extra brackets *do*?

**Comment by MTCam** [comment]
Is there a forthcoming fix for this?

**Comment by matthiasdiener** [comment]
> Is there a forthcoming fix for this?

Should be fixed already.

**Comment by MTCam** [comment]
Got it, merged with master to pull in the fix.

**Comment by MTCam** [comment]
@matthiasdiener Can you make anything of this zip failure thing? I don't see any errors reported in the details.

**Comment by MTCam** [comment]
I think the trouble might be that I am passing a relative path (./) to makezip.

File: install-pip-dependencies.sh [hunk 15]
```diff
@@ -1,11 +1,27 @@
...
 set -o nounset -o errexit
 
+origin=$(pwd)
+requirements_file="${1-mirgecom/requirements.txt}"
+install_location="${2-$origin}"
 
+if [ ! -d "$install_location" ]
+then
+    mkdir -p $install_location
+fi
```
Comment (matthiasdiener) [inline install-pip-dependencies.sh:21]:
```suggestion
mkdir -p $install_location
```

Comment (MTCam) [reply install-pip-dependencies.sh:21]: hehe

**Pull request merged**